### PR TITLE
Move AppWrapper API to workload.codeflare.dev group

### PR DESCRIPTION
**Pull request by astefanutti**
Part of project-codeflare/adr#7

Follows up project-codeflare/multi-cluster-app-dispatcher#591

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -21,6 +21,8 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3
 )
 
+replace sigs.k8s.io/custom-metrics-apiserver => sigs.k8s.io/custom-metrics-apiserver v1.25.1-0.20230306170449-63d8c93851f3
```
Comment (sutaakar) [inline go.mod:24]: What is this needed for?

Comment (astefanutti) [reply go.mod:24]: It's required by MCAD, as the compatible release of custom-metrics-adapter for k8s 1.26 is missing: https://github.com/project-codeflare/multi-cluster-app-dispatcher/blob/05b1af83014ce022d4d593b7af00cd9099d8d0b0/go.mod#L26

It has to be carried over here, as `replace` instructions are not inherited transitively.

It'll be removed once we upgrade to a newer version of k8s.

**Review by sutaakar — COMMENTED** [review]
/lgtm

**Review by sutaakar — COMMENTED** [review]
/lgtm

**Review by anishasthana — APPROVED** [review]
/lgtm

**Pull request merged**